openlibrary-wikidata-bot/jobs/sync_author_identifiers_from_wikidata.py

```diff
@@ -0,0 +1,210 @@
+    """
+    if not hasattr(author, "remote_ids"):
+        write_error(wd_id, author.olid, author.name, "openlibrary_author_remote_ids_failed_to_fetch", "", "")
+        return {}, -1
+    output = {**author.remote_ids}
+    # Count
+    update_count = 0
+    conflicts = 0
+    for identifier in WD_IDENTIFIERS.values():
+        if identifier in output and identifier in incoming_ids:
+            if output[identifier] != incoming_ids[identifier]:
+                conflicts = conflicts + 1
+                write_error(wd_id, author.olid, author.name, "openlibrary_wikidata_remote_id_collision", identifier, f'{{"ol": "{output[identifier]}", "wd": "{incoming_ids[identifier]}"}}')
+        elif identifier in incoming_ids and identifier not in output:
+            output[identifier] = incoming_ids[identifier]
+            update_count = update_count + 1
+    if conflicts > 0:
+        return author.remote_ids, -1
+    return output, update_count
+
+
+def consolidate_remote_author_ids(sql_path: str, dry_run: bool = True) -> None:
+    ol = OpenLibrary()
+
+    csv.field_size_limit(sys.maxsize)
+
+    # Read in the wikidata DB dump
+    with open(sql_path, mode="r", encoding="utf-8") as file:
+        next(file) # skip the header line
+        reader = csv.reader(file, delimiter="\t")  # Read as TSV
+        for row in reader:
+            if len(row) < 2:
+                continue
+            wikidata_col_raw = row[1]
+            
+            try:
+                parsed_wikidata_json = json.loads(wikidata_col_raw)
+                wd_json_top_level_fields = parsed_wikidata_json["statements"]
+
+                # Skip if no OL ID
+                if not ("P648" in wd_json_top_level_fields):
+                    continue
+
+                # Some of these WD entries match multiple OL authors. 
+                # What to do in that scenario?
+                # For now, flagging that as a problem
+
+                ol_ids = [
+                    obj["value"]["content"]
+                    for obj in wd_json_top_level_fields.get("P648", [])
+                    if validate_wikidata_key(obj, "P648") and re.fullmatch(r"OL\d+A", obj["value"]["content"])
+                ]
+
+                if len(ol_ids) == 0:
+                    continue
+                wd_id = parsed_wikidata_json["id"]
+
+                if len(ol_ids) > 1:
+                    for ol_id in ol_ids:
+                        author = ol.Author.get(ol_id)
+                        write_error(wd_id, ol_id, author.name, "multiple_openlibrary_authors_for_one_wikidata_row", "ol_id", f'[{",".join([f'"{val}"' for val in ol_ids])}]')
+                    continue
+                ol_id = ol_ids[0]
+
+                author = ol.Author.get(ol_id)
+
+                remote_ids = {}
+
+                # Check for just the identifiers we're interested in
+                for wd_p_value, remote_ids_key in WD_IDENTIFIERS.items():
```

Comment:
Trying to keep these names a bit more consistent. PID is common in Wikidata-land. In OL it seems like we use the word `name` for these

```suggestion
                for pid, ol_identifier_name in WD_IDENTIFIERS.items():
```